syntax = "proto3";
package proto;

import "FileInfo.proto";
import "OpenFile.proto";
///////TODO Clean up code, maybe seperate into different files
message Empty {
  
}

message StatusResponse {
  bool status = 1;
  string statusMessage = 2;
}



//////////////

//////////////////

enum RegionType {
  POINT = 0;
  LINE = 1;
  RECTANGLE = 2;
  CIRCLE = 3;
}

// message mask{
//   repeated bool mask = 1;
//   int32 width = 2;
//   int32 height = 3;
// }

message RegionDataRequest{
    string uuid = 1;
    RegionType regionType = 2;
    repeated int32 start = 3;
    repeated int32 count = 4;
}


//Change To Bytes
message RegionDataResponse{
  repeated float data = 1;
}


message SpectralProfileRequest {
  string uuid = 1;
  RegionType regionType = 2;
  int32 x = 3;
  int32 y = 4;
  int32 z = 5;
  int32 numPixels = 6;
  int32 width = 7;
  int32 height = 8;

}

//Change TO Bytes
message SpectralProfileResponse {
  repeated float data = 1;
  // bytes data = 1;
}

service H5ReaderServices{
  rpc CheckStatus(Empty) returns  (StatusResponse) {}
  //TODO Change to OpenFileACK
  rpc OpenFile(OpenFileRequest) returns (OpenFileACK) {}
  rpc CloseFile(FileCloseRequest) returns (StatusResponse) {}
  rpc GetFileInfo (FileInfoRequest) returns (FileInfoResponse) {}
  // rpc GetRegion (RegionDataRequest) returns (RegionDataResponse) {}
  // rpc GetSpectralProfile (SpectralProfileRequest) returns (SpectralProfileResponse) {}
  rpc GetRegionStream (RegionDataRequest) returns (stream RegionDataResponse) {}
  rpc GetSpectralProfileStream (SpectralProfileRequest) returns (stream SpectralProfileResponse) {}

}